### ISSUE TITLE
Error linting MySQL create table statements

### ISSUE THREAD

**Issue by kunagpal** [sqlfluff/sqlfluff#814]
## Expected Behaviour
The MySQL schema is parsed correctly, with lint issues pointed out where applicable.

## Observed Behaviour
Parsing fails, even with explicit `--dialect` and `--templater` settings.

### Schema
```sql
CREATE TABLE `foo` (
  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
```

### Options
```
--templater raw --dialect mysql
```

### Lint output
```
L:   1 | P:   1 |  PRS | Found unparsable section: 'CREATE TABLE `foo` (\n `id`
                       | int(11) unsi...'
```

### Parse output
```
[0](1, 1, 1)        |file:
[0](1, 1, 1)        |    statement:
[0](1, 1, 1)        |        unparsable:                                           !! Expected: 'statement'
[0](1, 1, 1)        |            raw:                                              'CREATE'
[6](1, 1, 7)        |            whitespace:                                       ' '
[7](1, 1, 8)        |            raw:                                              'TABLE'
[12](1, 1, 13)      |            whitespace:                                       ' '
[13](1, 1, 14)      |            raw:                                              '`foo`'
[18](1, 1, 19)      |            whitespace:                                       ' '
[19](1, 1, 20)      |            start_bracket:                                    '('
[20](1, 1, 21)      |            newline:                                          '\n'
[21](1, 2, 1)       |            whitespace:                                       '  '
[23](1, 2, 3)       |            raw:                                              '`id`'
[27](1, 2, 7)       |            whitespace:                                       ' '
[28](1, 2, 8)       |            raw:                                              'int'
[31](1, 2, 11)      |            start_bracket:                                    '('
[32](1, 2, 12)      |            raw:                                              '11'
[34](1, 2, 14)      |            end_bracket:                                      ')'
[35](1, 2, 15)      |            whitespace:                                       ' '
[36](1, 2, 16)      |            raw:                                              'unsigned'
[44](1, 2, 24)      |            whitespace:                                       ' '
[45](1, 2, 25)      |            raw:                                              'NOT'
[48](1, 2, 28)      |            whitespace:                                       ' '
[49](1, 2, 29)      |            raw:                                              'NULL'
[53](1, 2, 33)      |            whitespace:                                       ' '
[54](1, 2, 34)      |            raw:                                              'AUTO_INCREMENT'
[68](1, 2, 48)      |            comma:                                            ','
[69](1, 2, 49)      |            newline:                                          '\n'
[70](1, 3, 1)       |            whitespace:                                       '  '
[72](1, 3, 3)       |            raw:                                              'PRIMARY'
[79](1, 3, 10)      |            whitespace:                                       ' '
[80](1, 3, 11)      |            raw:                                              'KEY'
[83](1, 3, 14)      |            whitespace:                                       ' '
[84](1, 3, 15)      |            start_bracket:                                    '('
[85](1, 3, 16)      |            raw:                                              '`id`'
[89](1, 3, 20)      |            end_bracket:                                      ')'
[90](1, 3, 21)      |            newline:                                          '\n'
[91](1, 4, 1)       |            end_bracket:                                      ')'
[92](1, 4, 2)       |            whitespace:                                       ' '
[93](1, 4, 3)       |            raw:                                              'ENGINE'
[99](1, 4, 9)       |            raw:                                              '='
[100](1, 4, 10)     |            raw:                                              'InnoDB'
[106](1, 4, 16)     |            whitespace:                                       ' '
[107](1, 4, 17)     |            raw:                                              'DEFAULT'
[114](1, 4, 24)     |            whitespace:                                       ' '
[115](1, 4, 25)     |            raw:                                              'CHARSET'
[122](1, 4, 32)     |            raw:                                              '='
[123](1, 4, 33)     |            raw:                                              'utf8mb4'
[130](1, 4, 40)     |            whitespace:                                       ' '
[131](1, 4, 41)     |            raw:                                              'COLLATE'
[138](1, 4, 48)     |            raw:                                              '='
[139](1, 4, 49)     |            raw:                                              'utf8mb4_unicode_ci'
[157](1, 4, 67)     |    statement_terminator:                                     ';'
[158](1, 4, 68)     |    newline:                                                  '\n'

==== parsing violations ====
L:   1 | P:   1 |  PRS | Found unparsable section: 'CREATE TABLE `foo` (\n `id` int(11)
                       | unsi...'
```

### Fix output
```
==== finding fixable violations ====
==== no fixable linting violations found ====
```

## Version
sqlfluff, version 0.4.0

## Configuration
```
N/A, not using any configuration files
```
